### ISSUE TITLE
[NOT AN ISSUE] how to disconnect from the device when the app is killed

### ISSUE THREAD

**Issue by WanftMoon** [dotnet-bluetooth-le/dotnet-bluetooth-le#602]
To help us fix your issue, please provide the information in the below template. If something causes a crash, provide as much information as you can gather.
Just imagine: we do not know what you are doing! 

> Note: There is often little we can do without a minimal reproducible sample of the issue, so please provide that in a **standalone git repository** and link it here.

Hi, 

There is a specific where the user kills the app and the connection sometimes linger with the device. On android i managed to close the connection on OnDestroy, on iOS i'm trying to use WillTerminate to do this, but the method DisconnectDeviceAsync does not return and the SO kill the app within 5 seconds. Is this the place to do this or is there a trick on iOS for it? Or does the library already handles this scenario?

## Steps to reproduce

1. App connects to device

2. go to multi tasking and swipe to kill app

3. close the connection to the device

## Expected behavior
N/A

## Actual behavior
N/A

### Crashlog

N/A

## Configuration

**Version of the Plugin:** 2.1.3

**Platform:** iOS

**Device:** iPhone 8

**Comment by janusw** [comment]
> There is a specific where the user kills the app and the connection sometimes linger with the device. On android i managed to close the connection on OnDestroy, on iOS i'm trying to use WillTerminate to do this, but the method DisconnectDeviceAsync does not return and the SO kill the app within 5 seconds. Is this the place to do this or is there a trick on iOS for it? Or does the library already handles this scenario?

I think in principle those are the right places for disconnecting. I'm not sure why it doesn't work on iOS. Does the disconnection work if you trigger it earlier in the app?

Btw, at some point I had the problem that I could never see `OnDestroy` being called on Android, but I forgot the details.

**Comment by WanftMoon** [comment]
> I think in principle those are the right places for disconnecting. I'm not sure why it doesn't work on iOS. Does the disconnection work if you trigger it earlier in the app?

Yeah, the disconnect works when triggered from a view model or view controlller, but it doesn't from the app delegate (this happens only on iOS)
 
> Btw, at some point I had the problem that I could never see `OnDestroy` being called on Android, but I forgot the details.

OnDestroy may or may not be called, the best way (that i could find to have it working consistently) to ensure it is called is to have a Foreground service running.
